### PR TITLE
Bump deps

### PR DESCRIPTION
**Pull request by PsiACE**
- bump deps
- make test work
- make clippy happy

### CODE REVIEW

**Comment by PsiACE** [comment]
Hi @jonhoo , would you like to take some time to review this PR?

File: Cargo.toml [hunk 1]
```diff
@@ -23,16 +23,17 @@ codecov = { repository = "jonhoo/msql-srv", branch = "master", service = "github
 maintenance = { status = "experimental" }
 
 [dependencies]
-nom = "5"
-mysql_common = "0.22"
-byteorder = "1"
+nom = "7.0.0-alpha1"
```
Comment (jonhoo) [inline Cargo.toml:26]: I'd like to not depend on an alpha version — could you stick with 6 here please?

File: tests/async.rs [hunk 2]
```diff
@@ -102,10 +102,14 @@ where
...
         });
 
-        tokio::runtime::current_thread::block_on_all(
-            mysql_async::Conn::new(format!("mysql://127.0.0.1:{}", port)).and_then(|conn| c(conn)),
-        )
-        .unwrap();
+        // Create the runtime
+        let rt = tokio::runtime::Runtime::new().unwrap();
+
+        rt.spawn_blocking(move || {
```
Comment (jonhoo) [inline tests/async.rs:108]: Why does this need `spawn_blocking`?

File: src/value/utils.rs [hunk 3]
```diff
@@ -0,0 +1,254 @@
+#[cfg(test)]
```
Comment (jonhoo) [inline src/value/utils.rs:1]: I'm not sure I follow this file — what is it testing? It seems to also contain a bunch of implementation, but that's only used for tests? If this is an orthogonal change, would you mind making it in a separate PR?

**Review by jonhoo — COMMENTED** [review]
Sorry, I've been away on vacation :)

Overall this looks great, thanks for taking the time! I left some notes inline.

**Comment by jonhoo** [comment]
Also note that the Postgres version was already bumped by #24, so you may need to merge from master :)

File: src/value/utils.rs (hunk 3 above)
Comment (PsiACE) [reply src/value/utils.rs:1]: Here's some code to help the tests run correctly, migrated from the old mysql_common. I don't think this is considered orthogonal :(

File: Cargo.toml (hunk 1 above)
Comment (PsiACE) [reply Cargo.toml:26]: The current dependencies have been carefully selected to avoid dependency conflicts.

File: tests/async.rs (hunk 2 above)
Comment (PsiACE) [reply tests/async.rs:108]: I will remove the dependency on tokio. I checked it and we don't need tokio.

**Comment by PsiACE** [comment]
@jonhoo , thank you for your work, in fact I really like them. Our company's project [datafuse](https://github.com/datafuselabs/datafuse/) is using msql-srv  and the team is happy to participate in the maintenance.

File: Cargo.toml (hunk 1 above)
Comment (jonhoo) [reply Cargo.toml:26]: I'm not sure I follow — why is an alpha dependency necessary here?

File: tests/async.rs (hunk 2 above)
Comment (jonhoo) [reply tests/async.rs]: Oh, that's weird. I wasn't actually proposing you remove tokio. Instead, I was just questioning the use of `spawn_blocking` specifically. I would think that `rt.block_on` would be sufficient if you remove the call to `.wait()`. The `.wait()` is just a utility function inside the mysql library that does that blocking for you. But using `.wait()` is also okay if you prefer that, it's just a bit less idiomatic.

File: src/value/utils.rs (hunk 3 above)
Comment (jonhoo) [reply src/value/utils.rs:1]: If this comes from `mysql_common`, why move it in here? Does it test anything that actually relates to this crate? Why was it removed from `mysql_common` in the first place?

File: tests/async.rs (hunk 2 above)
Comment (PsiACE) [reply tests/async.rs]: Sorry, I will fix it later :( I was not thinking that much.

File: src/value/utils.rs (hunk 3 above)
Comment (PsiACE) [reply src/value/utils.rs:1]: This is because `mysql_common` removed functions like `write_bin_value`. In fact, this is just to ensure that the existing tests work properly.

It looks like `mysql_common` is now updated with serialization/deserialization.

File: Cargo.toml (hunk 1 above)
Comment (PsiACE) [reply Cargo.toml:26]: I also prefer the stable version, but unfortunately. If you upgrade `mysql_common`,  whether nom 5 or 6, there will be dependency conflicts.

File: tests/async.rs (hunk 2 above)
Comment (PsiACE) [reply tests/async.rs]: I guess we'll have to rely on `wait()` for now. Is it okay to change it to this?

```rust
        let rt = tokio::runtime::Runtime::new().unwrap();

        rt.block_on(async move {
            mysql_async::Conn::new(format!("mysql://127.0.0.1:{}", port))
                .and_then(|conn| c(conn))
                .wait()
                .unwrap()
        });
```

Or would you like to commit a patch telling me what to do? cc @jonhoo

File: Cargo.toml (hunk 1 above)
Comment (PsiACE) [reply Cargo.toml:26]: for nom 6: error: failed to select a version for `funty`
for nom 5, sorry, nom 5 is okay. But this will lag behind the major releases. If you want, we can go back to nom5.

Comment (jonhoo) [reply Cargo.toml:26]: Huh, that's very weird. What exactly is the error you get for `funty`?

File: tests/async.rs (hunk 2 above)
Comment (jonhoo) [reply tests/async.rs]: I was thinking:
```rust
rt.block_on(async move {
    mysql_async::Conn::new(format!("mysql://127.0.0.1:{}", port))
        .and_then(|conn| c(conn))
        .await.unwrap()
});
```
The `wait` method is just a helper method that `mysql_async` provides, but since we actually execute in async context here, we can do it properly and use `await`.

File: src/value/utils.rs (hunk 3 above)
Comment (jonhoo) [reply src/value/utils.rs:1]: Hmm, I don't think they actually removed those functions, they just renamed them. For example, decimal decoding is now in [`Decimal::parse_bytes`](https://docs.rs/mysql_common/0.27.4/mysql_common/binlog/decimal/struct.Decimal.html#method.parse_bytes). I think we should just switch to those methods instead of inlining the old implementation (which may have been replaced because it was buggy).

File: Cargo.toml (hunk 1 above)
Comment (PsiACE) [reply Cargo.toml:26]: nom 7 is now available. We can skip this.

**Comment by sundy-li** [comment]
After https://github.com/blackbeam/rust_mysql_common/pull/43 is merged.

**Comment by jonhoo** [comment]
I'll just add that I don't think we should need to have tests for `mysql_common` in _this_ crate. We should assume that it does "the right thing", and simply rely on that. Errors in its logic should be tested for in that crate instead, so maybe contribute these tests there?

**Comment by jonhoo** [comment]
Do we still then need to wait for https://github.com/blackbeam/rust_mysql_common/pull/43? I don't see a failing test, which suggests either that we do not, or that we should add a test to catch the issue represented by https://github.com/blackbeam/rust_mysql_common/pull/43.

**Comment by jonhoo** [comment]
Ah, no, I think we've talked past each other. What I was object to originally was the code introduced [here](https://github.com/jonhoo/msql-srv/pull/22/files/2dbb4e99c0e96b4536c9fb6da7b8bfa084104f10#diff-25de782ea8b52b529d451cfd6754668e8a30ec04e5f9bf2758f196991b82251d). The `rt!` tests should remain, since they test that _our_ encoding and decoding matches the MySQL client's decoding and encoding respectively. Those tests are not tests of `mysql_common`, they're tests of _our_ code under the assumption that `mysql_common` does the right thing.

**Comment by sundy-li** [comment]
> Do we still then need to wait for blackbeam/rust_mysql_common#43?

Yes, I've added the `rt!` tests and these are not passed.

There are 2 problems with `rust_mysql_common`

-   `ValueDeserializer<TextValue>` only return Value::Null and Value::Bytes,   [codes](https://github.com/blackbeam/rust_mysql_common/blob/master/src/value/mod.rs#L299-L314), this is reasonable since it's not enough to determine the type only with texts.
-  `ValueDeserializer<TextValue>` did not respect the `ColumnFlags::UNSIGNED_FLAG`

The author did not think it's buggy, I still did not figure out why.

**Comment by jonhoo** [comment]
Thinking some more about this, I wonder if what we should do is change `rt!` so that instead of directly serializing and deserializing the values, it constructs a full server and client, and then checks that if the server sends a value `v` in a column, the client produces a row with a value `v` of that same type. And also that if the client sends a value `v`, as, say, a prepared statement value, the server receives a value `v` of that same type. That way we should be testing the "whole" machinery, rather than whether the low-level interface matches (which it sounds like it doesn't any more).

What do you think?

**Comment by jonhoo** [comment]
This landed in #31.
